### PR TITLE
fix 5868 - make DefaultHostsFileEntriesResolverTest.java pass without relying on there being a localhost entry in the hosts file

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
Signed-off-by: radai-rosenblatt radai.rosenblatt@gmail.com

### CODE REVIEW

**Comment by normanmaurer** [comment]
@radai-rosenblatt please fix compiler error:

https://garage.netty.io/teamcity/viewLog.html?buildId=12748&buildTypeId=pulls_netty&tab=buildLog#_

We use java6 features :)

Also please sign the icla and update commit message to match our template:
http://netty.io/s/icla
http://netty.io/wiki/writing-a-commit-message.html

**Comment by radai-rosenblatt** [comment]
@normanmaurer - code fixed to only use java 6, icla signed and commit message expanded to match template :-)

File: resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java [hunk 1]
```diff
@@ -28,6 +32,21 @@
     @Test
     public void testLocalhost() {
         DefaultHostsFileEntriesResolver resolver = new DefaultHostsFileEntriesResolver();
+        //make sure there's _some_ localhost entry. put one if there isnt
```
Comment (normanmaurer) [inline resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java]: @radai-rosenblatt after thinking more about it maybe we should just use `Assume.*` for this test and revert the rest ?

Comment (radai-rosenblatt) [inline resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java]: ...maybe make the map field (DefaultHostsFileEntriesResolver.entries) non-final, replace it with a Mock in this test and assert that whatever string is searched for is lowercased (which is the real functionality being tested).

**Comment by radai-rosenblatt** [comment]
@normanmaurer I've reworked the patch. the test code now checks that host resolution is case insensitive more "directly", instead of relying on the existence of a localhost entry.

File: resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java [hunk 2]
```diff
@@ -19,16 +19,37 @@
...
+
+        Map<String, InetAddress> entriesMap = mock(Map.class);
+        Field entriesField = DefaultHostsFileEntriesResolver.class.getDeclaredField("entries");
+        entriesField.setAccessible(true);
+        entriesField.set(resolver, entriesMap);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        resolver.address("UPPERCASE");
+        resolver.address("uppercase");
```
Comment (Scottmitch) [inline resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java]: any reason for the host name change? consider reverting back to `LOCALHOST` and `localhost` ... awkward to read `uppercase` but its actually in lowercase.

Comment (radai-rosenblatt) [inline resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java]: was trying to drive home the point that this test isnt really about localhost specifically, its about resolution in general being case insensitive. reverted back to "LOCALHOST" vs "localhost".

**Comment by normanmaurer** [comment]
@radai-rosenblatt I would prefer if we could remove the usage of reflection honestly.y

**Comment by radai-rosenblatt** [comment]
@normanmaurer - I've fixed the patch to avoid reflection in the test (had to refactor the class being tested a bit)

**Comment by normanmaurer** [comment]
Cherry-Picked into 4.1 (4e2530c171edc98a94ea91fda1e988963b0cbf38)

**Comment by normanmaurer** [comment]
@radai-rosenblatt thanks a lot!

**Pull request closed without merging**